docs/use-cases.rst

```diff
@@ -0,0 +1,668 @@
+- Auditor
+
+Not all of these users interact with development tooling directly, or in the
+same ways. As such, these use cases should specify appropriately.
+
+This list may not be exhaustive, but should cover the vast majority of required
+behaviors.
+
+
+Saving artifacts
+----------------
+
+Users and tooling must be able to save new and update existing |Artifact|
+files.
+
+As a core behavior, tooling must facilitate the internal record-keeping of all
+known |ContractTypes| and |ContractInstances|, updating them appropriately
+when they change, or when knowledge of them changes.
+
+Beyond that, developers may also need to manually generate new artifacts for
+previously unknown contracts.
+
+Since a smart contract blockchain network is effectively a distributed global
+database, developers may write their applications for the express or implicit
+purpose of accessing other contract instances on that network.
+
+Developers should be able to leverage tooling to interface with these external
+contracts, either common, public libraries, or contract instances for other
+applications. Tooling should integrate well with code written outside a given
+project.
+
+Additionally, some applications may not use tooling's built-in deployment
+systems, and some |ContractInstances| may deploy others. Tooling should be
+able to account for the record-keeping in all these cases.
+
+.. uml::
+
+  left to right direction
+
+  :Smart Contract Developer: as :SmartContractDev:
+
+  rectangle Saving {
+    (Save contract type) as (SaveType)
+    (Save contract instance) as (SaveInstance)
+    (Save library instance) as (SaveLibrary)
+    (Save interface instance) as (SaveInterface)
+  }
+
+  SmartContractDev -- SaveType
+  SmartContractDev -- SaveInstance
+  SmartContractDev -- SaveLibrary
+  SmartContractDev -- SaveInterface
+
+Save contract type
+``````````````````
+
+Users must be able to save new |ContractTypes|, whether or not it has any
+corresponding |ContractInstances|.
+
+Save contract instance
+``````````````````````
+
+Users must be able to save new |ContractInstances| on a given network,
+identifying them as a known |ContractType|.
+
+Save library instance
+`````````````````````
+Similarly, users must be able to save records of externally-deployed
+|Libraries|. This is particularly useful, as many commonly-used libraries are
+already deployed on a majority of |Networks|, and users should not have to pay
+the extra gas just to accommodate tooling.
+
+Save interface instance
+```````````````````````
+
+In cases where source is not known for an external contract, users may need
+reference that |ContractInstance| by its interface (i.e. via Solidity's
+``interface`` mechanism).
+
+
+Compilation
+-----------
+
+At the core of smart contract development and deployment is the compilation
+of high-level languages to the underlying EVM |Bytecode|.
+
+Smart contract developers compile/recompile |Sources| many times throughout
+the entire lifecycle of development.
+
```

Comment:
`throughout the entire` is possibly pleonasm e.g. `she saw with her eyes`.

Reply:
ooh that's a three finney word right there